Safari Web Push (#4885)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/web/push_notifications/safari_mobile_push.md

---
nav_title: Safari Mobile Web Push
article_title: Safari Mobile Web Push
platform: Web
channel: push
page_order: 5
page_type: reference
description: "Learn how to integrate web push on your iOS and iPad Safari browsers."
search_rank: 3
---

# Web Push on Safari Mobile (iOS and iPadOS)

[Safari v16.4][safari-release-notes] adds support for mobile web push, which means you can now re-engage mobile users with Push Notifications on iOS and iPadOS.

This article will guide you through the steps required to set up mobile push for safari.

## Integration Steps

First, please read and follow our standard [web push integration guide][web-push-integration]. The following steps are only required to support web push on Safari for iOS and iPadOS support.

### Step 1. Create a Manifest File {#manifest}

A [Web Application Manifest][manifest-file] is a JSON file that controls how your website is presented when installed to a user's home screen.

For example, you can set the background theme color and icon that the [App Switcher][app-switcher] uses, whether it renders as full screen to resemble a native app, or whether the app should open in landscape or portrait mode.

> Create a new `manifest.json` file in your website's root directory, with the following mandatory fields. The full list of supported fields can be found [here][https://developer.mozilla.org/en-US/docs/Web/Manifest].

Reviewer: ```suggestion
Create a new `manifest.json` file in your website's root directory, with the following mandatory fields. The full list of supported fields can be found [here](https://developer.mozilla.org/en-US/docs/Web/Manifest).
```

```json
{
  "name": "your app name",
  "short_name": "your app name",
  "display": "fullscreen",
  "icons": [{
    "src": "favicon.ico",
    "sizes": "128x128",
  }]
}
```

### Step 2. Link the Manifest File {#manifest-link}

> Add the following `<link>` tag to your website's HTML pointing to where your manifest file is hosted.

Reviewer: might want to specifically say 

```suggestion
Add the following `<link>` tag to your website's `<head>` element pointing to where your manifest file is hosted.
```

so it doesn't end up buried in some local push-specific html that isn't loaded on the homepage

```html
<link rel="manifest" href="/manifest.json" />
```

### Step 3. Service Worker {#service-worker}

Your website must have a Service Worker file that imports the Braze service-worker library, as described in our [web push integration guide][service-worker].

### Step 4. Add To Homescreen {#add-to-homescreen}
![An iphone showing options to bookmark a website and save to the homescreen][add-to-homescreen-img]{: style="max-width:250px;float:right;"}

Unlike major browsers like Chrome and Firefox, you are not allowed to request push permission on Safari iOS/iPadOS unless your website has been added to the user's homescreen. 

The [Add to Homescreen][add-to-homescreen] feature lets users bookmark your website, adding your icon to their valuable homescreen real estate.

Once the app has been added to your homescreen you can now request push permission when the user takes an action (i.e. clicks a button) using the [`requestPushPermission`][requestPushPermission] method:

<br>

![A push prompt asking to "allow" or "don't allow" Notifications][safari-push-prompt-img]{: style="max-width:250px;float:right;"}

```typescript

> import * as braze from "@braze/web-sdk";
> 
> button.onclick = function(){
>     braze.requestPushPermission(() => {
>         console.log(`User accepted push prompt`);
>     }, (temporaryDeny) => {
>         console.log(`User temporarily dismissed push prompt`);
>     });
> };

Reviewer: can this be 

```suggestion
import { requestPushPermission } from "@braze/web-sdk";

button.onclick = function(){
    requestPushPermission(() => {
        console.log(`User accepted push prompt`);
    }, (temporary) => {
        console.log(`User ${temporary ? "temporarily dismissed" : "permanently denied"} push prompt`);
    });
};
```

?

Create a new `manifest.json` file in your website's root directory, with the following mandatory fields. 

```json
{
  "name": "your app name",
  "short_name": "your app name",
  "display": "fullscreen",
  "icons": [{
    "src": "favicon.ico",
    "sizes": "128x128",
  }]
}
```

The full list of supported fields can be found [here][manifest-file].

### Step 2. Link the Manifest File {#manifest-link}

Add the following `<link>` tag to your website's `<head>` element pointing to where your manifest file is hosted.

```html
<link rel="manifest" href="/manifest.json" />
```

### Step 3. Service Worker {#service-worker}

Your website must have a service worker file that imports the Braze service-worker library, as described in our [web push integration guide][service-worker].

### Step 4. Add To Homescreen {#add-to-homescreen}

![An iphone showing options to bookmark a website and save to the homescreen][add-to-homescreen-img]{: style="max-width:20%;float:right;"}

> Unlike major browsers like Chrome and Firefox, you are not allowed to request push permission on Safari iOS/iPadOS unless your website has been added to the user's homescreen. 

Reviewer 2: To confirm, Chrome or Firefox already allows users to request push permission on iOS/iPadOS?

Author: yes
